project/hash.go

```diff
@@ -11,13 +11,11 @@ import (
 
 // GetServiceHash computes and returns a hash that will identify a service.
 // This hash will be then used to detect if the service definition/configuration
-// have changed and needs to be rebuild.
-func GetServiceHash(service Service) string {
+// have changed and needs to be recreate.
+func GetServiceHash(name string, config *ServiceConfig) string {
 	hash := sha1.New()
 
-	io.WriteString(hash, fmt.Sprintln(service.Name()))
-
-	config := service.Config()
+	io.WriteString(hash, name)
```

Comment:
@dnephin Your right, @imikushin something happened here.  In my original PR getHash() took an image name.  You seem to have dropped that, so there is not more logic that checks the imageId has changed.  https://github.com/docker/libcompose/pull/16/files#diff-caac1491e4add415147af39b799d3bdaR258